src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs

```diff
@@ -604,6 +604,7 @@ private void WriteByteSlow(byte value)
             }
             else
             {
+                Debug.Assert(_writePos == _bufferSize);
```

Comment:
Requested here: https://github.com/dotnet/runtime/pull/51151#discussion_r612020237

Reply:
Changing to below suggestion in order to fix build error:
```suggestion
                Debug.Assert(_writePos == _bufferSize - 1);
```